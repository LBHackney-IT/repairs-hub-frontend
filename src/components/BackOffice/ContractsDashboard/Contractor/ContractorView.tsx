import { useQuery } from 'react-query'
import { useMemo, useState } from 'react'
import { fetchContracts } from '../../requests'

import Layout from '../../Layout'
import Spinner from '../../../Spinner'
import SorSearch from '../SorSearch'

import ContractListItems from '../Contract/ContractListItems'

interface ContractorViewProps {
  contractorReference: string
  contractorName: string
}

const ContractorView = ({
  contractorReference,
  contractorName,
}: ContractorViewProps) => {
  const [sorCode, setSorCode] = useState<string>(null)

  const {
    data: activeContracts,
    isLoading: activeContractsLoading,
    error: activeError,
  } = useQuery(
    [
      'activeContracts',
      { isActive: true, contractorReference: contractorReference },
    ],
    () => fetchContracts(true, contractorReference)
  )

  const {
    data: inactiveContracts,
    isLoading: inactiveContractsLoading,
    error: inactiveError,
  } = useQuery(
    [
      'inactiveContracts',
      { isActive: false, contractorReference: contractorReference },
    ],
    () => fetchContracts(false, contractorReference)
  )
  const {
    data: contractsWithSorCode,
    isLoading: sorContractsIsLoading,
    refetch: refetchSorContracts,
    error: sorContractsError,
  } = useQuery(
    ['sorContracts', contractorReference, sorCode?.toLocaleUpperCase()],
    () =>
      fetchContracts(null, contractorReference, sorCode?.toLocaleUpperCase()),
    {
      enabled: false,
    }
  )

  const descendingDateContractsWithSorCode = useMemo(
    () => (contractsWithSorCode ? [...contractsWithSorCode].reverse() : null),
    [contractsWithSorCode]
  )

  const activeContractsError = activeError as Error | null
  const inactiveContractsError = inactiveError as Error | null
  const contractsWithSorCodeError = sorContractsError as Error | null

  const relativeInactiveContracts = inactiveContracts?.filter(
    (contract) =>
      contract.terminationDate > '2020' &&
      contract.terminationDate < new Date().toISOString()
  )

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault()
    refetchSorContracts()
  }
  return (
    <Layout title={`${contractorName}`}>
      {activeContractsLoading ? (
        <Spinner />
      ) : (
        activeContracts && (
          <ContractListItems
            contracts={activeContracts}
            heading="Active contracts:"
            warningText={`No active contracts found for ${contractorName}.`}
            error={activeContractsError}
            page="contractor"
            testId="active"
          />
        )
      )}
      {inactiveContractsLoading ? (
        <Spinner />
      ) : (
        relativeInactiveContracts && (
          <ContractListItems
            contracts={relativeInactiveContracts}
            heading="Inactive contracts (from 2020):"
            warningText={`No inactive contracts found for ${contractorName}.`}
            error={inactiveContractsError}
            page="contractor"
            testId="inactive"
          />
        )
      )}
      <SorSearch
        searchHeadingText={'Check an SOR code'}
        searchLabelText={`Find out which ${contractorName} contracts an SOR code exists in`}
        sorCode={sorCode}
        setSorCode={setSorCode}
        isLoading={sorContractsIsLoading}
        error={contractsWithSorCodeError}
        handleSubmit={(e: React.FormEvent) => handleSubmit(e)}
        contracts={descendingDateContractsWithSorCode}
        contractorName={contractorName}
      />
    </Layout>
  )
}

export default ContractorView
