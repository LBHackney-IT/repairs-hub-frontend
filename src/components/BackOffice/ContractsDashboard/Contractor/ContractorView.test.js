import { render } from '@testing-library/react'

import ContractorView from './ContractorView'

import { mockActiveContracts, mockInactiveContracts } from '../mockContracts'

jest.mock('react-query', () => ({
  useQuery: () => ({
    data: mockActiveContracts,
    isLoading: false,
    error: null,
  }),
}))

jest.mock('react-query', () => ({
  useQuery: () => ({
    data: mockInactiveContracts,
    isLoading: false,
    error: null,
  }),
}))

describe('Contracts dashboard component', () => {
  it('should render the component', async () => {
    const { asFragment } = render(<ContractorView contractorReference="SYC" />)

    expect(asFragment()).toMatchSnapshot()
  })
})
