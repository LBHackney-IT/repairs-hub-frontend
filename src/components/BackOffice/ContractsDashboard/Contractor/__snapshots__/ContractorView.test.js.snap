// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contracts dashboard component should render component with sor code contracts displayed 1`] = `
<DocumentFragment>
  <a
    class="govuk-back-link lbh-back-link govuk-!-display-none-print"
    role="button"
  >
    Back
  </a>
  <h1
    class="lbh-heading-h1"
  >
    Sycous Limited SYC
  </h1>
  <h3
    class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
  >
    Active contracts:
  </h3>
  <ol
    data-test-id="active-contracts-list"
    style="display: grid; grid-template-columns: repeat(auto-fill, 10rem); gap: 1rem; list-style: none; padding: 0px; margin: 0px;"
  >
    <li
      data-id="0"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          126-126-1266
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £0.00
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           0
        </p>
        <p>
          Expired: 
          <span
            style="font-weight: 600;"
          >
            9 Aug 2025
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="1"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          127-127-1277
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £0.00
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           0
        </p>
        <p>
          Expires: 
          <span
            style="font-weight: 600;"
          >
            9 Jan 2026
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="2"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          128-128-1288
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £5,958.38
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           34
        </p>
        <p>
          Expires: 
          <span
            style="font-weight: 600;"
          >
            9 Jan 2026
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="3"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          129-129-1299
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £544,898.23
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           3441
        </p>
        <p>
          Expires: 
          <span
            style="font-weight: 600;"
          >
            9 Jan 2026
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="4"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          130-130-1300
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £573,623.97
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           3443
        </p>
        <p>
          Expires: 
          <span
            style="font-weight: 600;"
          >
            9 Jan 2026
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="5"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          131-131-1311
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £143,393.57
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           904
        </p>
        <p>
          Expires: 
          <span
            style="font-weight: 600;"
          >
            9 Jan 2026
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="6"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          132-132-1322
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £56.74
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           2
        </p>
        <p>
          Expires: 
          <span
            style="font-weight: 600;"
          >
            9 Jan 2026
          </span>
        </p>
      </div>
    </li>
  </ol>
  <h3
    class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
  >
    Inactive contracts (from 2020):
  </h3>
  <ol
    data-test-id="inactive-contracts-list"
    style="display: grid; grid-template-columns: repeat(auto-fill, 10rem); gap: 1rem; list-style: none; padding: 0px; margin: 0px;"
  >
    <li
      data-id="0"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          133-133-1388
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £5,958.38
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           20
        </p>
        <p>
          Expired: 
          <span
            style="font-weight: 600;"
          >
            11 May 2025
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="1"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          144-144-1444
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £5,958.38
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           20
        </p>
        <p>
          Expired: 
          <span
            style="font-weight: 600;"
          >
            11 May 2025
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="2"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          155-155-1555
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £5,958.38
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           20
        </p>
        <p>
          Expired: 
          <span
            style="font-weight: 600;"
          >
            11 May 2025
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="3"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          166-166-16666
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £5,958.38
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           20
        </p>
        <p>
          Expired: 
          <span
            style="font-weight: 600;"
          >
            8 Jul 2025
          </span>
        </p>
      </div>
    </li>
  </ol>
  <div>
    <section
      class="section"
    >
      <h3
        class="lbh-heading-h3"
      >
        Check an SOR code
      </h3>
      <div
        class="govuk-form-group lbh-form-group"
      >
        <form>
          <label
            class="govuk-label lbh-label"
            for="input-search"
          >
            Find out which Sycous Limited contracts an SOR code exists in
          </label>
          <input
            class="govuk-input lbh-input govuk-input--width-10"
            data-testid="input-search"
            id="input-search"
            name="search-name"
            type="text"
            value="ABC123"
          />
          <div
            class="govuk-form-group lbh-form-group"
          >
            <button
              class="govuk-button lbh-button"
              data-module="govuk-button"
              data-testid="submit-search"
              id="submit-search"
              type="submit"
            >
              Search
            </button>
          </div>
        </form>
      </div>
    </section>
    <h3
      class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
    >
      ABC123 found in the following Sycous Limited contracts
    </h3>
    <ol
      data-test-id="contract-list"
      style="display: grid; grid-template-columns: repeat(auto-fill, 10rem); gap: 1rem; list-style: none; padding: 0px; margin: 0px;"
    >
      <li
        data-id="0"
        style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
      >
        <div
          class="contract-details"
        >
          <h5
            style="white-space: nowrap;"
          >
            132-132-1322
          </h5>
          <p
            style="color: green;"
          >
            Active
          </p>
        </div>
      </li>
      <li
        data-id="1"
        style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
      >
        <div
          class="contract-details"
        >
          <h5
            style="white-space: nowrap;"
          >
            155-155-1555
          </h5>
          <p
            style="color: red;"
          >
            Expired: 11 May 2025
          </p>
        </div>
      </li>
    </ol>
  </div>
</DocumentFragment>
`;

exports[`Contracts dashboard component should render the component without displaying sor contracts 1`] = `
<DocumentFragment>
  <a
    class="govuk-back-link lbh-back-link govuk-!-display-none-print"
    role="button"
  >
    Back
  </a>
  <h1
    class="lbh-heading-h1"
  >
    Sycous Limited SYC
  </h1>
  <h3
    class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
  >
    Active contracts:
  </h3>
  <ol
    data-test-id="active-contracts-list"
    style="display: grid; grid-template-columns: repeat(auto-fill, 10rem); gap: 1rem; list-style: none; padding: 0px; margin: 0px;"
  >
    <li
      data-id="0"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          126-126-1266
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £0.00
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           0
        </p>
        <p>
          Expired: 
          <span
            style="font-weight: 600;"
          >
            9 Aug 2025
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="1"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          127-127-1277
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £0.00
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           0
        </p>
        <p>
          Expires: 
          <span
            style="font-weight: 600;"
          >
            9 Jan 2026
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="2"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          128-128-1288
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £5,958.38
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           34
        </p>
        <p>
          Expires: 
          <span
            style="font-weight: 600;"
          >
            9 Jan 2026
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="3"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          129-129-1299
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £544,898.23
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           3441
        </p>
        <p>
          Expires: 
          <span
            style="font-weight: 600;"
          >
            9 Jan 2026
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="4"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          130-130-1300
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £573,623.97
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           3443
        </p>
        <p>
          Expires: 
          <span
            style="font-weight: 600;"
          >
            9 Jan 2026
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="5"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          131-131-1311
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £143,393.57
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           904
        </p>
        <p>
          Expires: 
          <span
            style="font-weight: 600;"
          >
            9 Jan 2026
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="6"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          132-132-1322
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £56.74
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           2
        </p>
        <p>
          Expires: 
          <span
            style="font-weight: 600;"
          >
            9 Jan 2026
          </span>
        </p>
      </div>
    </li>
  </ol>
  <h3
    class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
  >
    Inactive contracts (from 2020):
  </h3>
  <ol
    data-test-id="inactive-contracts-list"
    style="display: grid; grid-template-columns: repeat(auto-fill, 10rem); gap: 1rem; list-style: none; padding: 0px; margin: 0px;"
  >
    <li
      data-id="0"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          133-133-1388
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £5,958.38
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           20
        </p>
        <p>
          Expired: 
          <span
            style="font-weight: 600;"
          >
            11 May 2025
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="1"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          144-144-1444
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £5,958.38
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           20
        </p>
        <p>
          Expired: 
          <span
            style="font-weight: 600;"
          >
            11 May 2025
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="2"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          155-155-1555
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £5,958.38
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           20
        </p>
        <p>
          Expired: 
          <span
            style="font-weight: 600;"
          >
            11 May 2025
          </span>
        </p>
      </div>
    </li>
    <li
      data-id="3"
      style="border: 5px solid #00664f; border-radius: 20px; box-sizing: border-box; padding: 1rem; display: flex; margin-top: 1rem; justify-content: center;"
    >
      <div
        class="contract-details"
      >
        <h5
          style="white-space: nowrap;"
        >
          166-166-16666
        </h5>
        <p>
          <span
            style="font-weight: 600;"
          >
            Sum of SORs:
          </span>
           £5,958.38
        </p>
        <p>
          <span
            style="font-weight: 600;"
          >
            SOR Count:
          </span>
           20
        </p>
        <p>
          Expired: 
          <span
            style="font-weight: 600;"
          >
            8 Jul 2025
          </span>
        </p>
      </div>
    </li>
  </ol>
  <div>
    <section
      class="section"
    >
      <h3
        class="lbh-heading-h3"
      >
        Check an SOR code
      </h3>
      <div
        class="govuk-form-group lbh-form-group"
      >
        <form>
          <label
            class="govuk-label lbh-label"
            for="input-search"
          >
            Find out which Sycous Limited contracts an SOR code exists in
          </label>
          <input
            class="govuk-input lbh-input govuk-input--width-10"
            data-testid="input-search"
            id="input-search"
            name="search-name"
            type="text"
            value=""
          />
          <div
            class="govuk-form-group lbh-form-group"
          >
            <button
              class="govuk-button lbh-button"
              data-module="govuk-button"
              data-testid="submit-search"
              disabled=""
              id="submit-search"
              type="submit"
            >
              Search
            </button>
          </div>
        </form>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
